lib/hammer_cli/defaults.rb

```diff
@@ -0,0 +1,120 @@
+            bad_input
+          else
+            namespace = ("HammerCLI" + option_plugin_name.split('_').collect!{ |w| w.downcase.capitalize }.join).constantize if option_plugin_name
+            param = option_param_val.nil? ? "from_server_"+namespace.to_s : option_param_val
+            Defaults.add_defaults_to_conf({option_param_name => param})
+            added_default_message({option_param_name.to_s => param.to_s})
+          end
+          rescue NameError => e
+          plugin_prob_message
+          rescue StandardError => e
+          file_not_found_message
+          HammerCLI::EX_CONFIG
+        end
+        HammerCLI::EX_OK
+      end
+    end
+
+    def added_default_message(options = {})
+      options.each do |key, val|
+        print_message("Added -" + key.to_s + " default-option with value -"+ val.to_s)
+      end
+    end
+
+    def plugin_prob_message(options = {})
+      print_message(_("Couldn't reach the plugin defaults class, eiter the plugin doesn't have defaults class or its unreachable."))
+    end
+
+    def bad_input(options = {})
+      print_message(_("You must specify a value or a plugin name."))
+    end
+
+    def file_not_found_message
+      print_message(_("Couldn't find any cli_config file in the recoginized paths"))
+    end
+
+    def list_all_defaults_message
+      HammerCLI::Settings.settings.each do |key,val|
+        if key.to_s.include?("defaults")
+          print_message('')
+          print_message(key.to_s+":")
+          print_message(YAML.dump(HammerCLI::Settings.settings[key]))
+        end
+      end
+    end
+
+    autoload_subcommands
+  end
+
+
+  class Defaults
+
+    def self.add_defaults_to_conf(options)
+      path = find_valid_conf_file
+      if path
+        new_file = YAML.load_file(path)
+        options.each do |key, value|
+          key = key.to_sym
+          new_file[:defaults] = {} unless new_file[:defaults]
+          value.to_s.to_i if value.is_a? Integer
+          new_file[:defaults][key] = value
+        end
+        File.open(path,'w') do |h|
+          h.write new_file.to_yaml
+        end
+      else
+        raise StandardError
+      end
+    end
+    #this method will be overriden by plugins who wish to have a defaults params.
+    def self.get_defaults(option)
+      unless HammerCLI::Settings.settings[:defaults].nil? || HammerCLI::Settings.settings[:defaults][option.to_sym].nil?
+        if HammerCLI::Settings.settings[:defaults][option.to_sym].to_s.include? "from_server"
+          value = HammerCLI::Settings.settings[:defaults][option.to_sym].split("_")[2].constantize::Defaults.get_defaults(option.to_sym)
+          value
+        else
+          value = HammerCLI::Settings.settings[:defaults][option.to_sym]
+          value
+        end
+      end
+    end
+
+    private
+    def self.find_valid_conf_file
+      path = nil
+      HammerCLI::Settings.path_history.each do |p|
+        path = p if p.include?("defaults.yml")
```

Comment:
Empty files are not recognized due to https://github.com/theforeman/hammer-cli/blob/master/lib/hammer_cli/settings.rb#L39